File: 1-two-sum/two-sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int *p;
    p = (int*)malloc(2 * sizeof(int));
    for (int i=0; i<numsSize; i++){
        for (int j=i+1; j<numsSize; j++){
            if (nums[i]+nums[j] == target){
                p[0] = i;
                p[1] = j;
                *returnSize = 2;
                break;
            }
        }
    }
    return p;
}

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] a = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int need = target - num;
            if (map.containsKey(need)) {
                a[0] = map.get(need);
                a[1] = i;
                return a;
            }
            map.put(nums[i], i);
        }
        return a;
    }
}